src/N3Parser.js

```diff
@@ -1040,8 +1043,16 @@ export default class N3Parser {
     this._lexer.tokenize(input, (error, token) => {
       if (error !== null)
         this._callback(error), this._callback = noop;
-      else if (this._readCallback)
-        this._readCallback = this._readCallback(token);
+      else if (this._readCallback) {
+        if (token.type === 'comment') {
```

Comment:
Expecting significant performance implication of checking this on every token; will likely require different approach.

Reply:
Another parse function, maybe called `parseWithComments`, would mitigate this I guess. Would that be acceptable?

Reply:
I'm thinking something akin to:
```
{
//...
  _processNextToken(token) {
    this._readCallback = this._readCallback(token);
  }
//...
}
```
which, in the case of comments, is monkey-patched with
```
this._processNextToken = function (token) {
  if (token.type === 'comment')
    this._commentCallback(token.value);
  else
    this._readCallback = this._readCallback(token);
}
```
or similar.

Possibly instead of monkey-patching, it _could_ be a boolean check in `_processNextToken` for whether or not comment mode is enabled.

Reply:
I’ve implemented monkey-patching